rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -117,29 +122,28 @@ public BookKeeper(Configuration conf)
             throw new TException(e);
         }
         endBlock = setCorrectEndBlock(endBlock, fileLength, remotePath);
-        List<Location> blocksInfo = new ArrayList<>((int) (endBlock - startBlock));
+        List<BlockLocation> blockLocations = new ArrayList<>((int) (endBlock - startBlock));
         int blockSize = CacheConfig.getBlockSize(conf);
 
         for (long blockNum = startBlock; blockNum < endBlock; blockNum++) {
             totalRequests++;
             long split = (blockNum * blockSize) / splitSize;
-
             if (md.isBlockCached(blockNum)) {
-                blocksInfo.add(Location.CACHED);
+                blockLocations.add(new BlockLocation(Location.CACHED, blockSplits.get(split)));
                 cachedRequests++;
             }
             else {
-                if (localSplits.contains(split)) {
-                    blocksInfo.add(Location.LOCAL);
+                if (blockSplits.get(split).equalsIgnoreCase(nodes.get(currentNodeIndex))) {
```

Comment:
Potential improvement, store Node indices too i.e. split to node index map and compare indices here instead of strings (nodenames). Leave a comment here, we can revisit and see what perf gain we get